### PR TITLE
health: track a bunch more stuff

### PR DESCRIPTION
**Pull request by bradfitz**
WIP

Just plumbing raw signals into the health package currently. Not analyzing health yet.

### CODE REVIEW

File: control/controlclient/auto.go [hunk 1]
```diff
@@ -179,7 +179,10 @@ func NewNoStart(opts Options) (*Client, error) {
 
 }
 
-func (c *Client) onHealthChange(key string, err error) {
+func (c *Client) onHealthChange(key health.ErrorKey, err error) {
+	if key == health.KeyOverall {
```
Comment (danderson) [inline control/controlclient/auto.go:183]: Why does overall health not trigger a map restart? Comment?

File: health/health.go [hunk 2]
```diff
@@ -162,14 +183,51 @@ func SetIPNState(state string, wantRunning bool) {
...
-	// For staticcheck for now:
+	if ipnState == "" {
+		// Don't check yet.
+		return
+	}
+	setLocked(KeyOverall, overallErrorLocked())
+}
+
+func overallErrorLocked() error {
+	if ipnState != "Running" || !ipnWantRunning {
```
Comment (danderson) [inline health/health.go:204]: Why is this an error? "correctly stopped" isn't an error state, is it?

**Review by danderson — COMMENTED** [review]
drive-by thoughts on draft.

**Comment by bradfitz** [comment]
I didn't realize this was still open.

This is an old PR. It was already merged elsewhere for 1.6.

We can go over the code as checked in sometime if you want, though.

**Comment by danderson** [comment]
I _thought_ it looked familiar-but-different :p . All good, when we do some pairing we can talk about it (not that I have complaints, just some thoughts of more checks and such)

**Pull request closed without merging**